### PR TITLE
Add flag `--output-json` for  `info` command

### PR DESCRIPTION
**Pull request by lean-apple**
In the wake of https://github.com/paritytech/cargo-contract/pull/993

Closes https://github.com/paritytech/cargo-contract/issues/796

### CODE REVIEW

**Comment by lean-apple** [comment]
Current state :
Screenshot 2023-03-10 at 10 52 51

**Comment by lean-apple** [comment]
I still need to add a test for the flag and to prepare the global tests for the upcoming flags but open to comments.

File: crates/cargo-contract/src/cmd/info.rs [hunk 1]
```diff
@@ -94,15 +112,34 @@ impl InfoCommand {
...
+    storage_item_deposit: u128,
+}
 
-    fn print_and_format_contract_info(info: ContractInfo) {
-        let convert_trie_id = hex::encode(info.trie_id.0);
-        name_value_println!("TrieId:", format!("{}", convert_trie_id));
-        name_value_println!("Code hash:", format!("{:?}", info.code_hash));
-        name_value_println!("Storage items:", format!("{:?}", info.storage_items));
+impl InfoToJson {
+    pub fn to_json(&self) -> Result<String> {
```
Comment (cmichi) [inline crates/cargo-contract/src/cmd/info.rs:126]: Why `pub`?

Comment (cmichi) [inline crates/cargo-contract/src/cmd/info.rs:126]: For `pub` functions there should be comments.

File: crates/cargo-contract/src/cmd/info.rs [hunk 2]
```diff
@@ -67,7 +70,22 @@ impl InfoCommand {
 
             match info_result {
                 Some(info_result) => {
-                    InfoCommand::print_and_format_contract_info(info_result);
+                    let output_type = match self.output_json {
+                        true => OutputType::Json,
+                        false => OutputType::HumanReadable,
+                    };
```
Comment (ascjones) [inline crates/cargo-contract/src/cmd/info.rs:76]: It is redundant to define and construct this enum. can just do `if self.output_json`.

I can see that it is copied from `build`, but in that case it is passed down to other commands via `ExecuteArgs` so makes more sense. Whereas in this case the conditional is evaluated inline.

File: crates/cargo-contract/src/cmd/info.rs (hunk 1 above)
Comment (ascjones) [reply crates/cargo-contract/src/cmd/info.rs:126]: IMO this method does not need to exist, can just do `serde_json::to_string_pretty(info_to_json)?` inline.

File: crates/cargo-contract/src/cmd/info.rs (hunk 2 above)
Comment (lean-apple) [reply crates/cargo-contract/src/cmd/info.rs:76]: Yep I hesitate to do this in this simple way

File: crates/cargo-contract/src/cmd/info.rs (hunk 1 above)
Comment (lean-apple) [reply crates/cargo-contract/src/cmd/info.rs:123]: Comments Added - as discussed with @ascjones after it is ok like this atm

File: crates/cargo-contract/src/cmd/info.rs [hunk 3]
```diff
@@ -50,6 +50,9 @@ pub struct InfoCommand {
         default_value = "ws://localhost:9944"
     )]
     url: url::Url,
+    /// Export the call output as JSON.
```
Comment (cmichi) [inline crates/cargo-contract/src/cmd/info.rs:53]:
```suggestion
    /// Export the instantiate output in JSON format.
```

This is a user-facing text, we should use the same description as other sub-commands do for`--output-json`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 - Add `info` command - [#993](https://github.com/paritytech/cargo-contract/pull/993)
+- Add `--output-json` flag for `info` command - [#1007](https://github.com/paritytech/cargo-contract/pull/1007)
```
Comment (cmichi) [inline CHANGELOG.md:11]:
```suggestion
```

**Review by cmichi — APPROVED** [review]
Thanks, Lea!

**Pull request merged**